Authentication/AuthenticationComponentsFactory.swift

```diff
@@ -306,23 +335,32 @@ extension SignupComponentsFactory {
         return SignupCredentialsValidator()
     }
     
-    public func createSignupViewModel<SessionService: SessionServiceType>(withSessionService sessionService: SessionService,
-                                      credentialsValidator: SignupCredentialsValidator,
-                                      configuration: SignupViewConfigurationType) -> SignupViewModelType {
-        return SignupViewModel(sessionService: sessionService,
-                               credentialsValidator: credentialsValidator,
-                               usernameEnabled: configuration.usernameEnabled,
-                               passwordConfirmationEnabled: configuration.passwordConfirmationEnabled)
+    public func createSignupViewModelFactory<SessionService: SessionServiceType>(withSessionService sessionService: SessionService,
```

Comment:
I would move the `withSessionService` one line below like in the other functions.